account_mtd/models/mtd_hello_world.py

```diff
@@ -0,0 +1,456 @@
+                menu_id = self.env.ref('account_mtd.submenu_mtd_hello_world')
+                return werkzeug.utils.redirect(
+                    '/web#id={id}&view_type=form&model=mtd.hello_world'.format(id=record_id) +
+                    '&menu_id={menu}&action={action}'.format(menu=menu_id.id, action=action.id)
+                )
+            return True
+    
+    @api.multi
+    def get_user_authorisation(self):
+        # get the action id and menu id and store it in the tracker table
+        # if we get an error on authorisation or while exchanging tokens we need to use these to redirect.
+        action = self.env.ref('account_mtd.action_mtd_hello_world')
+        menu_id = self.env.ref('account_mtd.submenu_mtd_hello_world')
+
+        # Update the information in the api tracker table
+        self._logger.info("(Step 1) Get authorisation")
+        tracker_api = self.env['mtd.api_request_tracker']
+        tracker_api = tracker_api.create({
+            'user_id': self._uid,
+            'api_id': self.api_name.id,
+            'api_name': self.api_name.name,
+            'endpoint_id': self.id,
+            'request_sent': True,
+            'action': action.id,
+            'menu_id': menu_id.id,
+        })
+
+        redirect_uri = "{}/auth-redirect".format(self.hmrc_configuration.redirect_url)
+        state = ""
+        # State is optional
+        if self.hmrc_configuration.state:
+            state = "&state={}".format(self.hmrc_configuration.state)
+        # scope needs to be percent encoded
+        scope = urllib.parse.quote_plus(self.scope)
+        authorisation_url_prefix = "https://test-api.service.hmrc.gov.uk/oauth/authorize?"
+        self._logger.info("(Step 1) Get authorisation - authorisation URI used:- {}".format(authorisation_url_prefix))
+        authorisation_url = (
+            "{url}response_type=code&client_id={client_id}&scope={scope}{state}&redirect_uri={redirect}".format(
+                url=authorisation_url_prefix,
+                client_id=self.hmrc_configuration.client_id,
+                scope=scope,
+                state=state,
+                redirect=redirect_uri
+            )
+        )
+        self._logger.info(
+            "(Step 1) Get authorisation - authorisation URI " +
+            "used to send request:- {url}".format(url=authorisation_url)
+        )
+        response = requests.get(authorisation_url, timeout=3)
+        # response_token = json.loads(req.text)
+        self._logger.info(
+            "(Step 1) Get authorisation - received response of the request:- {response}".format(response=response)
+        )
+        if response.ok:
+            return {'url': authorisation_url, 'type': 'ir.actions.act_url', 'target': 'self', 'res_id': self.id}
+        else:
+            response_token = json.loads(response.text)
+            error_message = (
+                "Date {date}     Time {time} \n".format(date=datetime.utcnow().date(), time=datetime.utcnow().time())
```

Comment:
Do you want to display times to users in UTC?

Reply:
yes in the time zone they are in right? That's what you suggested in the previous pr.

Reply:
No they're in Europe/London.  When BST starts these strings will be 1 hour out.

The ORM's date/datetime/time fields work in UTC (which is the context in which I suggested using utcnow()), but when formatting a string for humans you need to use local time.